website/pages/docs/internals/security.mdx

```diff
@@ -0,0 +1,322 @@
+    Using something like [Vault](/docs/vault-integration/index.html) to
+    create and manage dynamic, rotated credentials is highly recommended to
+    prevent secrets from being easily exposed within the [job
+    specification](/docs/job-specification/index.html)
+    itself which may be leaked into version control or otherwise be accidently
+    stored on disk on an operator’s local machine. It is also possible to
+    [integrate with Vault’s PKI secret engine](/guides/security/vault-pki-integration.html)
+    to automatically generate and renew dynamic, unique X.509 certificates for
+    each Nomad node with a short
+    [TTL](https://en.wikipedia.org/wiki/Time_to_live).
+
+* **[Running without Root](https://groups.google.com/forum/#!topic/nomad-tool/pSyMwC_FSFA)** -
+    Certain features of Nomad can be used without needing to run the Nomad agent
+    server or client as the `root` user. Instead you can granularly assign the
+    appropriate capabilities in various ways for your Nomad agents. For example:
+    Nomad servers only require access to the data directory; it is possible to
+    use Nomad to orchestrate Docker containers by adding a non-root `nomad` user
+    to the `docker` group to access the [default unix
+    socket](https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option).
+
+* **Containers with Sandbox Runtimes** - In some situations, such as running
+    untrusted code as a service, it may be worth considering using different
+    container runtimes such as [gVisor](https://gvisor.dev/) or [Kata
+    Containers](https://katacontainers.io/). These types of runtimes provide
+    sandboxing features which help prevent raw access to the underlying shared
+    kernel for other containers and the Nomad client agent itself.
+
+* **[Disable Unused Drivers](/docs/configuration/client#driver-blacklist)** -
+    Each driver provides different degrees of isolation, and bugs may allow
+    unintended privilege escalation. If a task driver is not needed, you can
+    disable it to reduce risk.
+
+* **Linux Security Modules** - Use of security modules that can be directly
+    integrated into operating systems such as AppArmor, SElinux, and Seccomp on
+    both the Nomad hosts and applied to containers for an extra layer of
+    security. Seccomp profiles are able to be passed directly to containers
+    using the
+    **[security_opt](/docs/drivers/docker.html#security_opt)**
+    parameter available in the default [Docker
+    driver](/docs/drivers/docker.html).
+
+* **[Service Mesh](https://www.hashicorp.com/resources/service-mesh-microservices-networking)** -
+    Integrating service mesh technologies such as
+    **[Consul](https://www.consul.io/)** can be extremely useful for limiting
+    and efficiently load balancing network connectivity within a cluster.
+
+### Threat Model
+
+The following are parts of the Nomad threat model:
+
+* **Nomad agent-to-agent communication** - Transport encryption for
```

Comment:
I think `User-to-agent communication (HTTP)` would make a good followup to this and describe how that's protected by mTLS and ACL tokens.
```suggestion
* **Nomad agent-to-agent communication (RPC)** - Transport encryption for
```

Not sure about specifying "HTTP" vs "RPC" ... might be a needlessly confusing implementation detail.

Reply:
umm.. what would go under "User-to-agent communication" is currently covered under the subsequent sections "Tampering of data in transit" and "Access to data without authentication or authorization".  Any suggestions for how to restructure them?